src/classes/RD2_ValidationService.cls

```diff
@@ -266,6 +281,28 @@ public with sharing class RD2_ValidationService {
         }
     }
 
+    /**
+    * @description Validates RD status is mapped to a valid state
+    * @param rd The Recurring Donation to validate
+    * @return void
+    */
+    @TestVisible
+    private void validateStatus(npe03__Recurring_Donation__c rd) {
+        if (String.isBlank(statusMapper.getState(rd.Status__c))) {
+            String errorMessage = String.join(
```

Comment:
How about simpler `System.Label.RD2_StatusMustBeMapped + '  ' + System.Label.RD2_StatusMustBeMappedSolution ` without using `List` allocation to join strings instead?